contracts/ParameterStore.sol

```diff
@@ -17,48 +21,77 @@ contract ParameterStore is AccessController {
     /// example for _archToLevRatio: If each arch is worth 1000 lvUSD, set this to 1000
     uint256 internal _archToLevRatio = 1 ether; // meaning 1 arch is equal 1 lvUSD
 
-    constructor(address admin) AccessController(admin) {}
+    modifier onlyGovernor() {
+        require(hasRole(GOVERNOR_ROLE, msg.sender), "Caller is not Governor");
+        _;
+    }
 
-    function init(address treasuryAddress) external initializer onlyAdmin {
+    function init(address treasuryAddress, address admin) external {
+        require(_initlized == false, "Cant re init contract");
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        _grantRole(GOVERNOR_ROLE, admin);
         _treasuryAddress = treasuryAddress;
+        _initlized = true;
+    }
+
+    /// TODO : Move access control to a simple lib
+
+    function setGovernor(address newGovernor) external {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Caller is not an Admin");
+        _grantRole(GOVERNOR_ROLE, newGovernor);
+    }
+
+    function revokeGovernor(address governor) external {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Caller is not an Admin");
+        _revokeRole(GOVERNOR_ROLE, governor);
+    }
+
+    function setAdmin(address newAdmin) external {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Caller is not an Admin");
+        _grantRole(DEFAULT_ADMIN_ROLE, newAdmin);
+    }
+
+    function revokeAdmin(address admin) external {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Caller is not an Admin");
```

Comment:
Should we make this also a modifier `onlyAdmin` ? Since we have a TODO to move to its own library it will make it easier to integrate with other contracts

Reply:
all the revoke and set/add roles are publicly accessible with an internal guard